esp-hal/src/i2c/master/mod.rs

```diff
@@ -110,7 +110,12 @@ pub enum Error {
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[non_exhaustive]
-pub enum ConfigError {}
+pub enum ConfigError {
+    /// Provided bus frequency is invalid for the current configuration.
+    InvalidFrequency,
```

Comment:
We should figure out how we can actually tell *what* is wrong - would printing warnings/errors make sense, if we don't want to overcomplicate the error variants?

Reply:
logging is probably a good trade-off - we could add s.th. like the possible frequency rang here to the error variant but I can't think of a case where that is useful to the user (other than logging / printing it during development) 🤔

Reply:
Maybe it should go under #2839 too?
I mean, I of course can do some research about it and implement it, but also, we don't do any (?) validation of frequency so far (IDF too though). Maybe it'll be better to cover that in terms of validation task and then return that error with needed limits?